### PR TITLE
recursive-patterns(14) Improve definite-assignment behavior in error cases for switch and is-pattern

### LINKED ISSUE #9154: Unexpected ERR_UseDefViolation error
```
    void Test5()
    {
        Dummy((System.Func<object, object, bool>) ((o1, o2) => o1 is int x5 && 
                                                               o2 is int x5 && 
                                                               x5 > 0));
    }
```

Observed:

```
    // (33,74): error CS0128: A local variable named 'x5' is already defined in this scope
    //                                                                o2 is int x5 && 
    Diagnostic(ErrorCode.ERR_LocalDuplicate, "x5").WithArguments("x5").WithLocation(33, 74),
    // (34,64): error CS0165: Use of unassigned local variable 'x5'
    //                                                                x5 > 0));
    Diagnostic(ErrorCode.ERR_UseDefViolation, "x5").WithArguments("x5").WithLocation(34, 64)
```

### PR DESCRIPTION
**Pull request by gafter**
Fixes #9154

@agocke @cston Please review this small stand-alone change to reduce cascaded diagnostics.
/cc @dotnet/roslyn-compiler

### CODE REVIEW

**Comment by gafter** [comment]
The code doesn't compile because of a typo.

**Comment by gafter** [comment]
@cston Can you please review this small stand-alone change to reduce cascaded diagnostics?

File: src/Compilers/CSharp/Test/Emit/Attributes/AttributeTests.cs [hunk 1]
```diff
@@ -6577,12 +6577,10 @@ public class A : Attribute
...
 
             CreateCompilation(source).VerifyDiagnostics(
-                // (7,31): error CS0182: An attribute argument must be a constant expression, typeof expression or array creation expression of an attribute parameter type
-                // [A(new object[] { default(E), default(E) })]
-                Diagnostic(ErrorCode.ERR_BadAttributeArgument, "default(E)").WithLocation(7, 31),
                 // (7,19): error CS0182: An attribute argument must be a constant expression, typeof expression or array creation expression of an attribute parameter type
                 // [A(new object[] { default(E), default(E) })]
-                Diagnostic(ErrorCode.ERR_BadAttributeArgument, "default(E)").WithLocation(7, 19));
+                Diagnostic(ErrorCode.ERR_BadAttributeArgument, "default(E)").WithLocation(7, 19)
+                );
```
Comment (cston) [inline src/Compilers/CSharp/Test/Emit/Attributes/AttributeTests.cs]: Please test errors in two separate arguments. This should result in two errors.
```
public A(object[] a, object[] b) { }
...
[A(new object[] { default(E) }, new object[] { default(E) })]
...
```

Comment (gafter) [reply src/Compilers/CSharp/Test/Emit/Attributes/AttributeTests.cs]: Test added.

**Comment by gafter** [comment]
@cston I believe I have addressed all of your comments. Please let me know if you feel we are suppressing too many attribute argument diagnostics.

**Comment by gafter** [comment]
@cston I have reverted the change regarding attribute arguments. Is this PR good now?

**Pull request merged**